hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -39,65 +34,30 @@ public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
     }
 
-    if (shouldRebuildTraceEventsGraph || shouldRebuildTraceEntitiesGraph) {
-      Instant start = Instant.now();
-      if (shouldRebuildTraceEventsGraph) {
-        cachedGraph.reCreateTraceEventsGraph(trace);
-      } else {
-        cachedGraph.reCreateTraceEntitiesGraph(trace);
-      }
-      if (LOG.isDebugEnabled()) {
-        LOG.debug(
-            "Time taken in building TraceEventsGraph, duration_millis:{} for tenantId:{}",
-            Duration.between(start, Instant.now()).toMillis(),
-            trace.getCustomerId());
-      }
-      cachedTraceThreadLocal.set(StructuredTrace.newBuilder(trace).build());
-      cachedGraphThreadLocal.set(cachedGraph);
-      debugGraph("Case: Partially building the graph.", cachedGraph, trace);
-      return cachedGraph;
-    }
-
     debugGraph("Case: Not building the graph.", cachedGraphThreadLocal.get(), trace);
     return cachedGraph;
   }
 
   private static void debugGraph(
       String logPrefix, StructuredTraceGraph graph, StructuredTrace trace) {
-    if (null != graph
-        && (null == graph.getTraceEntitiesGraph() || null == graph.getTraceEventsGraph())) {
+    if (null != graph && null == graph.getTraceEventsGraph()) {
       LOG.info(
           logPrefix
-              + "StructuredTraceGraph is not built correctly, trace {}, Is events graph non-null: {}."
-              + " Is entities graph non-null: {}",
+              + "StructuredTraceGraph is not built correctly, trace {}, Is events graph non-null:"
+              + " {}.",
           trace,
-          (null != graph.getTraceEventsGraph()),
-          (null != graph.getTraceEntitiesGraph()));
+          (null != graph.getTraceEventsGraph()));
 
       // build the graph again and check
       StructuredTraceGraph tempGraph = new StructuredTraceGraph(trace);
       LOG.info(
-          logPrefix
-              + "Recreating StructuredTraceGraph. Is events graph non-null: {}."
-              + " Is entities graph non-null: {}",
-          (null != tempGraph.getTraceEventsGraph()),
-          (null != tempGraph.getTraceEntitiesGraph()));
+          logPrefix + "Recreating StructuredTraceGraph. Is events graph non-null: {}.",
+          (null != tempGraph.getTraceEventsGraph()));
 
       tempGraph.reCreateTraceEventsGraph(trace);
       LOG.info(
-          logPrefix
-              + "Recreating events graph. Is events graph non-null: {}."
-              + " Is entities graph non-null: {}",
-          (null != tempGraph.getTraceEventsGraph()),
-          (null != tempGraph.getTraceEntitiesGraph()));
-
-      tempGraph.reCreateTraceEntitiesGraph(trace);
-      LOG.info(
-          logPrefix
-              + "Recreating entities graph. Is events graph non-null: {}."
-              + " Is entities graph non-null: {}",
-          (null != tempGraph.getTraceEventsGraph()),
-          (null != tempGraph.getTraceEntitiesGraph()));
+          logPrefix + "Recreating events graph. Is events graph non-null: {}.",
```

Comment:
Do we need this now - `reCreateTraceEventsGraph`? Same would have happen above, right?